### DISCUSSION TITLE
how can I know the agent offline in version-8.7.0

### DISCUSSION THREAD

**Discussion by HotDogOvO** [apache/skywalking#10922] [Q&A] [answered]
oap version: 8.7.0
agent version: 8.7.0
in oap server, how can I know the agent offline, I can only know the agent online in [ ManagementServiceHandler.java ] rightnow,
if I can't get it in oap server, is there any fields to distinguish agent status in db ?

**Comment by wu-sheng** [comment]
I don't know what do you mean `how`? `How` from where? CLI, query, database or something?

**Reply by HotDogOvO** [reply]
in query, I used the graphql interface getAllServices, I found that it doesn't have a status field to distinguish whether a service is offline, So I looked at the oap server code, I did not find out what the oap did after the heartbeat of the agent was disconnected

**Reply by wu-sheng** [reply, marked as answer]
Service doesn't go offline, but the instance will.

**Reply by wu-sheng** [reply]
Service only disappears when TTL is reached, and no new traffic related to this service is detected anymore.

**Reply by HotDogOvO** [reply]
> Service doesn't go offline, but the instance will.

oh I was negligent, thank you
